### ISSUE TITLE
Saving as Pull Request only works if one has writing privileges to that repository. Can it be made to work when one does not?

### ISSUE THREAD

**Issue by ilovan** [cwrc/CWRC-GitWriter#66]
I realize this may be more complicated because CWRC_Writer would also have to fork the repo and then send the pull request, but I think this is something that would be useful to people who want to do any crown sourcing. 

Is it doable @ajmacdonald and @lucaju ?

**Comment by ajmacdonald** [comment]
Yes that's doable.

**Comment by ilovan** [comment]
great, thanks @ajmacdonald . 
Then please go ahead and implement. 

I will leave the two of you to sort who does what (between the NERVE Edited/Unedited andthis one).

**Comment by lucaju** [comment]
I was doing some research about this. 

**Part 1**
My understanding is that to open a PR, the modified file (commit) must be in a different branch. So, to open a PR from CWRC-GitWriter, it is necessary to first create a branch to store the changes, then open a PR (in fact, this is what the current code does).

This works perfectly for the file owner. However, to open a PR in someone else's public repository, the user must have writing privileges to the head or the source branch [https://octokit.github.io/rest.js/v18#pulls-create](https://octokit.github.io/rest.js/v18#pulls-create).

This means that a user can access and read from public repos, but not save or open a PR without permission. In fact, this makes a lot of sense. Imagine how easy would be to flood public repos with PRs. My understanding is that the owner must invite a user to collaborate, then this user will have access to the file and will be able to open a PR.

**Part 2**
Finally, I understood what Mihaela was proposing and Andrew asserted that it is doable.
First, check if the user has writing privileges.
Without permission writing privileges, CWRC-WRITER fork the repo (from where the file was loaded) to the user's GitHub account.
Then, from forked repo, it opens a PR in the original repo.
Is this it @ilovan ?

**Comment by lucaju** [comment]
Yes, it is possible to create a Fork on behave of the user [https://octokit.github.io/rest.js/v18#repos-create-fork](https://octokit.github.io/rest.js/v18#repos-create-fork).

There is just one thing that worries me. Note on the API: " Forking a Repository happens asynchronously. You may have to wait a short period of time before you can access the git objects. If this takes longer than 5 minutes, be sure to contact GitHub Support or GitHub Premium Support.”

Everything is ASYNC. But for them to say no more than 5 min worries me that the request can take a long time to process. Maybe because it is copying files from one user to another? In that case, we should warn the user that this action might take longer to complete.

Well, there is only one way to see it… let’s do it.

**Comment by lucaju** [comment]
After a few days exploring the code, doing some updates, and trying a few things, I have a hard-coded prototype of this feature. But before implementing it, I'd like to suggest a quick revision on the saving steps (diagram attached). 

<img width="2201" alt="Save Document" src="https://user-images.githubusercontent.com/1254739/90396681-9fc41280-e064-11ea-8df7-55940c5d3885.png">
